### DISCUSSION TITLE
Quickly reapply recent tags and titles

### DISCUSSION THREAD

**Discussion by N3C2L** [madprops/grasshopper#35] [Ideas]
I would love to apply the same tag or title to other tabs more faster than to retype them again. I am thinking of a list command: If command executed on an actual tab spawn a list with the 10 recent tags or titles to apply on the tab. This would be aweseome.

**Comment by madprops** [comment]
I'll try adding some below the prompts for tags and titles. Thing is, I currently have no way to know when a title or tag was applied. So I can't sort them. But I can remember last value used before the extension closes, like on a sidebar.

**Comment by madprops** [comment]
Ok I can save objects to tabs, so I might be able to save tags with dates.

**Comment by madprops** [comment]
What I did for now is add autocomplete suggestions on tab prompts.

**Comment by N3C2L** [comment]
Thanks. The recently tag thing is not a must. I'd be happy too if I could have a static favorite tag/title list to quickly apply from.

**Comment by N3C2L** [comment]
If I use the tags as a to-do list, then tags should be replacable. Because if a tab has the tag "Today", I would like to click on this tag and replace it with something else like "Monday". Hence I think a "Replace Tag" command would come handy. Additionally If I click the tag, the tag list should spawn to choose a tag for replacing the actual tag. That would be great. This command could be named something like "Replace Tag Menu".

**Comment by madprops** [comment]
Already got it working, the question is, if I replace "Today" with "Monday", should it replace it only on the current tab, or on all tabs with the tag "Today"?

**Comment by N3C2L** [comment]
Thanks. I would prefer to replace the tag on the current tab only.

**Comment by madprops** [comment]
So this is what I'm going to do. I'm gonna add a command called `Replace Tag` which replaces a tag among all tabs, I think that's useful. But on the taglist, the `Edit` mode will act in the way you want, to edit a current tag. So you can set the taglist mode to Edit.

**Comment by N3C2L** [comment]
Thanks. What about this: When you use the "Edit Tag" command then autospawn the already used tags to quickly paste them in the edit field. The same tags that would appear while typing (autosuggestion) in the edit field.

**Comment by madprops** [comment]
I just uploaded it. How it works is that it brings up the prompt with the tag auto filled, then you can edit it. `v2028`.

**Comment by madprops** [comment]
Just set the Taglist Mode to `Edit`. Also right clicking the taglist items is supported to show the menu. And middle click removes the tags.

**Comment by N3C2L** [comment]
Thanks, but I thought about something different without typing with the keyboard: What about a "favorite tag menu": In this menu you could add your favorite tags. When right click on a tag you then could chose a favorite tag.

**Comment by madprops** [comment]
I don't like the idea of having to maintain such a list, manually by the user. What I might do is store used tags on some list, ordered by last recent date. And then have a button in the prompt to select from those 10 or so recent tags.

**Comment by N3C2L** [comment]
Yes, that would be nice to have. But it would be great if these stored used tags would auto popup while executing the "edit tag" command.

**Comment by madprops** [comment]
So a button to summon a list is not enough, how would you expect them to be displayed?

**Comment by madprops** [comment]
![tag2345](https://github.com/madprops/grasshopper/assets/206498/816c0801-3ff1-4681-b3f5-a2a74b1f4da2)

This works nicely. Also can use multiple times to add multiple tags. Or you can just write them.

I'll upload soon.

**Comment by N3C2L** [comment]
Thanks. Yeah, sorry I should have express my idea better. I meant after executing the "edit title" command it should auto popup the stored tags to prevent another mouseclick (could have been an option). But maybe this is more convenient and faster for me: Display these stored used tags only in a separate command. If you then chose one of these stored tags it closes the list immediately and apply the selected tag to the tab. Advantage would here less clicks (and it work same like a menu when right click, for example choosing colors).

**Comment by madprops** [comment]
I could add a command like that for `Add Tag` maybe called `Add Tag List`. You spawn a list and pick a tag and it adds it to the item(s). Or `Pick Tag`.

**Comment by N3C2L** [comment]
That would be great.

**Comment by madprops** [comment]
Which will be filterable since my `NeedContext` library supports filtering by typing something.

Uploading soon.

**Comment by madprops** [comment]
Added `Pick Tag`. on `v2029`. And other bunch of stuff.

**Comment by N3C2L** [comment]
Thank you so much. Appreciate the patience you have with me.

**Comment by madprops** [comment]
No problem.
In the process I make the code better by rethinking/refactoring stuff. So it's all good. 
Just can't keep adding lots of new features as it is already at a decent complexity level.

**Comment by N3C2L** [comment]
I see. I won't ask for too much in the near future. Just a few things regarding tags and titles if you don't mind? You decide the time to implement it, no hurry. This would  be amazing.

I would be glad about a "Pick Title" command (similar function like the "Pick Tag" command).

Following thing would help me: If "Pick Tag" command is executed after left click an existing tag of a tab, replace the tag with the one from the "Pick Tag" interface.

Please add the "Add Tag" command to the "Pick Tag" interface. Maybe a button with a plus symbol anywhere in this interface? Because if the tag is not in the list I could quickly add another tag with this new button.

**Reply by madprops** [reply]
1) I can add another list for recently used title like I do with tags, ok

2) You're talking about clicking an item of the tag list and spawning the pick tag to replace it, correct?
I don't understand this one.

3) Actually what if I remove Pick Tag .. and just make Add Tag spawn the list automatically? Since it already has the list functionality.

Question: Is the extra click on the smiley face on Add Tags too much? It would simplify to just have that prompt interface as is, and remove Pick Tag.

What I'm gonna do is remove Pick Tag, and make the list auto spawn when Adding tags, not on edit.

Edit Title won't autospawn the list but the list will be available with the smiley. I don't want a Pick Title. But I'll make it auto submit when picking one.

**Comment by N3C2L** [comment]
Thanks, I agree with all of your thoughts.
Yes, you are right about the second point with your comment "clicking an item of the tag list and spawning the pick tag to replace it, correct?".  Hence I suggested another command which autospawns the list to replace a tag.

**Comment by madprops** [comment]
What I'm gonna do is add a smiley list to the edit tag prompt, it should have one.

**Comment by N3C2L** [comment]
Thats ok. + If "edit tag" command is executed, highlight the selected tag in the edit field + autospawn the list of tags (to reduce clicks). All in one command executed.

**Comment by N3C2L** [comment]
I tested your updates and they work fine, thanks a bunch. Now I can tag and filter my tags as a to-do list (tags for days, priority, postpone, etc.).  And picking titles that I already used is timesaving. I've asked for a lot in the last few days and I'm going to back off a bit. If I find something useful or a bug I will post it here again on github.
